File: android/java/v2/AmpliApp/app/src/main/java/com/example/ampliapp/MainActivity.java
package com.example.ampliapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Button;

import com.amplitude.ampli.*;

import java.util.HashMap;
import java.util.Map;

public class MainActivity extends AppCompatActivity {
    private final String userId = "ampli-java-user-id";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btnIdentify = this.findViewById(R.id.btn_identify);
        btnIdentify.setOnClickListener(v -> {
            Ampli.getInstance().identify(userId, Identify.builder()
                    .requiredNumber(42.0)
                    .build());
        });

        Button btnSetGroup = this.findViewById(R.id.btn_set_group);
        btnSetGroup.setOnClickListener(v -> {
            Ampli.getInstance().getClient().setGroup("test group", "android-java-ampli");
        });

        Button btnGroupIdentify = this.findViewById(R.id.btn_group_identify);
        btnGroupIdentify.setOnClickListener(v -> {
            Map<String, Object> groupProperties = new HashMap<String, Object>() {{
                put("requiredBoolean", true);
            }};
            Ampli.getInstance().getClient().groupIdentify("test group", "android-java-ampli", groupProperties);
        });

        Button btnEventWithOptionalProperties = this.findViewById(R.id.btn_optional_properties);
        btnEventWithOptionalProperties.setOnClickListener(v -> {
            Ampli.getInstance().track(EventWithOptionalProperties.builder()
                    .optionalBoolean(true)
                    .build(), null);
        });

        Button btnEventWithAllProperties = this.findViewById(R.id.btn_all_properties);
        btnEventWithAllProperties.setOnClickListener(v -> {
            Ampli.getInstance().eventWithAllProperties(EventWithAllProperties.builder()
                    .requiredArray(new String[]{"I'm", "required"})
                    .requiredBoolean(false)
                    .requiredEnum(EventWithAllProperties.RequiredEnum.ENUM1)
                    .requiredInteger(42)
                    .requiredNumber(1.23)
                    .requiredString("Hi!")
                    .build());
        });

        Button btnOtherEvents = this.findViewById(R.id.btn_other_events);
        btnOtherEvents.setOnClickListener(v -> {
            Ampli.getInstance().eventMaxIntForTest(EventMaxIntForTest.builder()
                    .intMax10(9)
                    .build());

            Ampli.getInstance().eventNoProperties();

            Ampli.getInstance().eventWithConstTypes();

            Ampli.getInstance().eventWithArrayTypes(EventWithArrayTypes.builder()
                    .requiredBooleanArray(new Boolean[]{true, false, true})
                    .requiredEnumArray(new String[]{"enum1"})
                    .requiredNumberArray(new Double[]{1.1, 2.2, 3.3})
                    .requiredObjectArray(new Object[]{1, "a", true})
                    .requiredStringArray(new String[]{"a", "bc", "def"})
                    .build());

            Ampli.getInstance().eventObjectTypes(EventObjectTypes.builder()
                    .requiredObject("abc")
                    .requiredObjectArray(new Object[]{1, "a", true})
                    .build());

            Ampli.getInstance().eventWithEnumTypes(EventWithEnumTypes.builder()
                    .requiredEnum(EventWithEnumTypes.RequiredEnum.REQUIRED_ENUM_2)
                    .build());

            Ampli.getInstance().eventWithOptionalArrayTypes(EventWithOptionalArrayTypes.builder()
                    .optionalBooleanArray(new Boolean[]{false, true})
                    .build());

            Ampli.getInstance().eventWithDifferentCasingTypes(EventWithDifferentCasingTypes.builder()
                    .enumCamelCase(EventWithDifferentCasingTypes.EnumCamelCase.ENUMCAMELCASE)
                    .enumPascalCase(EventWithDifferentCasingTypes.EnumPascalCase.ENUMPASCALCASE)
                    .enumSnakeCase(EventWithDifferentCasingTypes.EnumSnakeCase.ENUM_SNAKE_CASE)
                    .enumWithSpace(EventWithDifferentCasingTypes.EnumWithSpace.ENUM_WITH_SPACE)
                    .propertyWithCamelCase("property with camel case")
                    .propertyWithPascalCase("property with pascal case")
                    .propertyWithSnakeCase("property with snake case")
                    .propertyWithSpace("property with space")
                    .build());

            Ampli.getInstance().eventWithTemplateProperties(EventWithTemplateProperties.builder()
                    .requiredEventProperty("event property")
                    .requiredTemplateProperty("template property")
                    .optionalEventProperty(1.23)
                    .build());
        });
    }
}

File: android/kotlin/v2/AmpliApp/app/src/main/java/com/example/ampliapp/MainActivity.kt
package com.example.ampliapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import com.amplitude.ampli.*

class MainActivity : AppCompatActivity() {
    private val userId: String = "ampli-kotlin-user-id"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val btnIdentify = findViewById<Button>(R.id.btn_identify)
        btnIdentify.setOnClickListener {
            ampli.identify(userId, Identify(
                requiredNumber = 42.0,
            ))
        }

        val btnSetGroup = findViewById<Button>(R.id.btn_set_group)
        btnSetGroup.setOnClickListener {
            ampli.setGroup("test group", "android-kotlin-ampli")
        }

        val btnGroupIdentify = findViewById<Button>(R.id.btn_group_identify)
        btnGroupIdentify.setOnClickListener {
            ampli.groupIdentify("test group", "android-kotlin-ampli", Group(requiredBoolean = true))
        }

        val btnEventWithOptionalProperties = findViewById<Button>(R.id.btn_optional_properties)
        btnEventWithOptionalProperties.setOnClickListener {

            ampli.track(EventWithOptionalProperties(
                optionalBoolean = true,
            ))
        }

        val btnEventWithAllProperties = findViewById<Button>(R.id.btn_all_properties)
        btnEventWithAllProperties.setOnClickListener {
            ampli.eventWithAllProperties(
                requiredNumber = 1.23,
                requiredArray = arrayOf("I'm", "required"),
                requiredBoolean = false,
                requiredEnum = EventWithAllProperties.RequiredEnum.ENUM_1,
                requiredInteger = 42,
                requiredString = "Hi!"
            )
        }

        val btnOtherEvents = findViewById<Button>(R.id.btn_other_events)
        btnOtherEvents.setOnClickListener {
            ampli.eventMaxIntForTest(
                intMax10 = 9
            )

            ampli.eventNoProperties()

            ampli.eventWithConstTypes()

            ampli.eventWithArrayTypes(
                requiredBooleanArray = arrayOf(true, false, true),
                requiredNumberArray = arrayOf(1.1, 2.2, 3.3),
                requiredStringArray = arrayOf("a", "bc", "def"),
                requiredObjectArray = arrayOf(1, "a", true)
            )

            ampli.eventObjectTypes(
                requiredObject = 3,
                requiredObjectArray = arrayOf(1, "a", true)
            )

            ampli.eventWithEnumTypes(
                requiredEnum = EventWithEnumTypes.RequiredEnum.REQUIRED_ENUM_2
            )

            ampli.eventWithOptionalArrayTypes(
                optionalBooleanArray = arrayOf(false, true)
            )

            ampli.eventWithDifferentCasingTypes(
                enumCamelCase = EventWithDifferentCasingTypes.EnumCamelCase.ENUM_CAMEL_CASE,
                enumPascalCase = EventWithDifferentCasingTypes.EnumPascalCase.ENUM_PASCAL_CASE,
                enumSnakeCase = EventWithDifferentCasingTypes.EnumSnakeCase.ENUM_SNAKE_CASE,
                enumWithSpace = EventWithDifferentCasingTypes.EnumWithSpace.ENUM_WITH_SPACE,
                propertyWithCamelCase = "property with camel case",
                propertyWithPascalCase = "property with pascal case",
                propertyWithSnakeCase = "property with snake case",
                propertyWithSpace = "property with space"
            )
        }
    }
}